packages/core/utils/src/serializer.js

```diff
@@ -16,37 +17,160 @@ export function registerSerializableClass(name: string, ctor: Class<*>) {
   ctorToName.set(ctor, name);
 }
 
-export function serialize(object: any): string {
-  return JSON.stringify(object, (key, value) => {
-    let serialized;
-    if (value && typeof value.serialize === 'function') {
-      // If the object has a serialize method, call it
-      serialized = value.serialize();
-    } else {
-      serialized = value;
+export function unregisterSerializableClass(name: string, ctor: Class<*>) {
+  if (nameToCtor.get(name) === ctor) {
+    nameToCtor.delete(name);
+  }
+
+  if (ctorToName.get(ctor) === name) {
+    ctorToName.delete(ctor);
+  }
+}
+
+function shallowCopy(object: any) {
+  if (object && typeof object === 'object') {
+    if (Array.isArray(object)) {
+      return [...object];
     }
 
-    // Add a $$type property with the name of this class, if any is registered.
-    if (
-      value &&
-      typeof value === 'object' &&
-      typeof value.constructor === 'function'
-    ) {
-      let type = ctorToName.get(value.constructor);
-      if (type != null) {
-        return {
-          $$type: type,
-          value: {...serialized}
-        };
+    if (object instanceof Map) {
+      return new Map(object);
+    }
+
+    if (object instanceof Set) {
+      return new Set(object);
+    }
+
+    return Object.create(
+      Object.getPrototypeOf(object),
+      Object.getOwnPropertyDescriptors(object)
+    );
+  }
+
+  return object;
+}
+
+function mapObject(object: any, fn: (val: any) => any, preOrder = false) {
```

Comment:
`function mapObject<T, U>(object: T, fn: (val: T) => U, preOrder = false) {` maybe?

the generic types without a constraint should behave like `mixed`, which is probably all you need, and can eliminate the `any`s here.

Reply:
I experimented but couldn't get it to work... :/ Not really sure the value of it anyway - it really could be anything.